packages/esbuild-plugin/src/plugins.ts

```diff
@@ -46,11 +41,11 @@ export const wrapPlugins = (build: PluginBuild, context: string) => {
 const getNewBuildObject = (
     build: PluginBuild,
     pluginName: string,
-    context: string
+    context: string,
 ): PluginBuild => {
     const newBuildObject: any = Object.assign({}, build);
-    for (const fn of Object.values(FN_TO_WRAP)) {
-        newBuildObject[fn] = async (opts: any, cb: any) => {
+    for (const fn of FN_TO_WRAP) {
+        newBuildObject[fn as keyof PluginBuild] = async (opts: any, cb: any) => {
```

Comment:
See my other comment, you should be able to remove this `as`
```suggestion
        newBuildObject[fn] = async (opts: any, cb: any) => {
```